src/main/java/teammates/ui/webapi/action/SearchStudentsAction.java

```diff
@@ -28,6 +28,13 @@ public void checkSpecificAccessControl() {
         if (!userInfo.isInstructor && !userInfo.isAdmin) {
             throw new UnauthorizedAccessException("Instructor or Admin privilege is required to access this resource.");
         }
+        String entity = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
```

Comment:
If entity type equals to something else, we might have unexpected flow in both `execute()` and `checkSpecificAccessControl()`

Reply:
Added throw exception if entities do not match exactly

Reply:
Don't forget test case :P